### PR TITLE
Improve custom figure example

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary
Just wanted to do a bit style cleanup, but ended up rewriting the whole example. :smile:

I didn't like about the original example that it's not a reasonable application. Setting a fixed position `Text` as title is not quite suitable because it is not contained in any layout. If anything, one should use `Figure.suptitle`. But defining a subclass for this would be overkill. Using `fig.suptitle('my title')` explicitly is easier than `plt.figure(FigureClass=MyFigure, title='my title')`.

### CODE REVIEW

File: examples/subplots_axes_and_figures/custom_figure_class.py [hunk 1]
```diff
@@ -1,27 +1,55 @@
...
-        """
-        custom kwarg figtitle is a figure title
-        """
+    def __init__(self, *args, watermark=None, **kwargs):
         super().__init__(*args, **kwargs)
-        self.text(0.5, 0.95, figtitle, ha='center')
+
+        if watermark is not None:
+            bbox = dict(boxstyle='square', lw=3, ec='gray',
+                        fc=(0.9, 0.9, .9, .5), alpha=0.5)
```
Comment (anntzer) [inline examples/subplots_axes_and_figures/custom_figure_class.py]: I don't really care whether you use 0.9 or .9, but at least be consistent :) (and same with 0.5 here and below)

File: examples/subplots_axes_and_figures/custom_figure_class.py [hunk 2]
```diff
@@ -1,27 +1,55 @@
 """
 ===================
-Custom Figure Class
+Custom Figure class
```
Comment (anntzer) [inline examples/subplots_axes_and_figures/custom_figure_class.py]: subclass?

File: examples/subplots_axes_and_figures/custom_figure_class.py [hunk 3]
```diff
@@ -1,27 +1,55 @@
...
 ===================
 
-You can pass a custom Figure constructor to figure if you want to derive from
-the default Figure.  This simple example creates a figure with a figure title.
+You can pass a custom Figure class to `.pyplot.figure` if you want to change
```
Comment (anntzer) [inline examples/subplots_axes_and_figures/custom_figure_class.py]: subclass?

File: examples/subplots_axes_and_figures/custom_figure_class.py (hunk 2 above)
Comment (timhoffm) [reply examples/subplots_axes_and_figures/custom_figure_class.py]: Maybe just "Subclassing Figure"?

Comment (anntzer) [reply examples/subplots_axes_and_figures/custom_figure_class.py]: fine with me too.

Comment (timhoffm) [reply examples/subplots_axes_and_figures/custom_figure_class.py]: Any preference? The second one is shorter the first one is more precise.

Comment (anntzer) [reply examples/subplots_axes_and_figures/custom_figure_class.py]: "Custom Figure subclasses"?

**Pull request merged**